hls4ml/model/graph.py

```diff
@@ -514,47 +516,52 @@ def insert_node(self, node, before=None, input_idx=0):
                 new_graph[node.name] = node
 
         self.graph = new_graph
-        self._update_model_outputs()
 
     def remove_node(self, node, rewire=True):
         """Remove a node from a graph.
 
-        By default, this function can connect the outputs of previous node to the input of next one.
-        Note that when removing a leaf node `rewire` should be set to `False`.
+        By default, this function can connect the outputs of previous
+        node to the input of next one. If the removed node has multiple
+        inputs/outputs tensors, an exception is raised.
 
         Args:
-            node (Layer): The node to remove
-            rewire (bool, optional): If `True`, connects the outputs of the previous node
-                to the inputs of the next node
+            node (Layer): The node to remove rewire (bool, optional):
+            Deprecated, no effect
```

Comment:
Would marking this as deprecated according to the [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) work? Would be nice to see